romancal/outlier_detection/outlier_detection_step.py

```diff
@@ -1,37 +1,202 @@
+        exptype = single_model.meta.exposure.type
+        self.check_input()
+
+        # check for TSO models first
+        if exptype == "WFI_IMAGE":
+            # imaging with resampling
+            detection_step = outlier_registry["imaging"]
+            pars["resample_suffix"] = "i2d"
+        else:
+            self.log.error(
+                "Outlier detection failed for unknown/unsupported ",
+                f"exposure type: {exptype}",
+            )
+            self.valid_input = False
+
+        if not self.valid_input:
+            if self.input_container:
+                for model in self.input_models:
+                    model.meta.cal_step.outlier_detection = "SKIPPED"
+            else:
+                self.input_models.meta.cal_step.outlier_detection = "SKIPPED"
+            self.skip = True
+            return self.input_models
+
+        self.log.debug(f"Using {detection_step.__name__} class for outlier_detection")
+        reffiles = {}
 
-    def process(self, input):
-        input_model = rdm.open(input, lazy_load=False)
+        # Set up outlier detection, then do detection
+        step = detection_step(self.input_models, reffiles=reffiles, **pars)
+        step.do_detection()
 
-        # No reference files
-        # reference_file_model = {}
-        # Do the outlier detection
-        # output_model = outlier_detection.do_correction(
-        #     input_model,
-        #     reference_file_model,
-        # )
+        state = "COMPLETE"
+        if self.input_container:
+            if not self.save_intermediate_results:
+                self.log.debug(
+                    "The following files will be deleted since save_intermediate_results=False:"
+                )
+            for model in self.input_models:
+                model.meta.cal_step.outlier_detection = state
+                if not self.save_intermediate_results:
+                    #  Remove unwanted files
+                    crf_path = self.make_output_path(basepath=model.meta.filename)
+                    #  These lines to be used when/if outlier_i2d files follow output_dir
+                    #  crf_file = os.path.basename(crf_path)
+                    #  outlr_path = crf_path.replace(crf_file, outlr_file)
+                    outlr_file = model.meta.filename.replace("cal", "outlier_i2d")
+                    blot_path = crf_path.replace("crf", "blot")
+                    median_path = blot_path.replace("blot", "median")
 
-        output_model = input_model
+                    for fle in [outlr_file, blot_path, median_path]:
+                        if os.path.isfile(fle):
+                            os.remove(fle)
+                            self.log.debug(f"    {fle}")
+        else:
+            self.input_models.meta.cal_step.outlier_detection = state
+        return self.input_models
 
-        # Close the input and reference files
-        input_model.close()
+    def check_input(self):
```

Comment:
As mentioned above, this method may not be needed if the registry is removed.

Reply:
We kept the method but refactored its content to check if the input is a valid `ModelContainer` or not.